packages/aws-cdk/lib/api/util/sdk.ts

```diff
@@ -207,19 +207,29 @@ export class SDK implements ISDK {
     }
     config.customUserAgent = userAgent;
 
-    // https://aws.amazon.com/blogs/developer/using-the-aws-sdk-for-javascript-from-behind-a-proxy/
-    options.proxyAddress = options.proxyAddress || httpsProxyFromEnvironment();
-    options.caBundlePath = options.caBundlePath || caBundlePathFromEnvironment();
+    const proxyAddress = options.proxyAddress || httpsProxyFromEnvironment();
+    const caBundlePath = options.caBundlePath || caBundlePathFromEnvironment();
 
-    if (options.proxyAddress) { // Ignore empty string on purpose
-      debug('Using proxy server: %s', options.proxyAddress);
-      httpOptions.proxy = options.proxyAddress;
+    if (proxyAddress && caBundlePath) {
+      throw new Error(`Cannot specify Proxy (${proxyAddress}) and CA Bundle (${caBundlePath}) at the same time`);
+      // Maybe it's possible after all, but I've been staring at
```

Comment:
If I understand correctly, there is a [use case](https://github.com/aws/aws-cdk/issues/5743#issuecomment-574059497) for using a proxy and a CA bundle, no?

Reply:
Probably. 

1) We didn't have CA certs before so not adding the feature is not a regression from that respect.

2) This PR is fixing the regression of proxy support working at all. Adding support for CAs while using the ProxyAgent can be a feature request that we deal with a later date. Have you seen the source code linked? I really don't want to add more risk by trying to figure out what is happening there that at this time.

Reply:
Fair enough. so maybe we can improve the error message to read:

```
At the moment, it is impossible to use a proxy with a custom CA bundle (see issue https://github...).
```